### PR TITLE
Add MaxLangVersion

### PR DESCRIPTION
**Pull request by agocke**
Will be consumed by tooling to recommend LangVersion upgrades

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
How does this relate to #37018?

File: src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets [hunk 1]
```diff
@@ -50,8 +50,9 @@
     </PropertyGroup>
 
     <PropertyGroup>
+      <MaxLangVersion Condition="'$(MaxLangVersion)' == ''">7.3</MaxLangVersion>
```
Comment (jasonmalinowski) [inline src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets]: https://github.com/dotnet/project-system/pull/5028 is defining a property "MaxSupportedLangVersion" -- is that what you intended to put here?

Comment (333fred) [reply src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets]: This property in general doesn't seem very good. It's the `MaxLangVersion`, but this is only true of .net framework apps and older .net core apps? This doesn't seem correct.

Comment (agocke) [reply src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets]: Yeah, MaxSupportedLangVersion is what I meant. I will update.

**Comment by agocke** [comment]
@jasonmalinowski This just adds the property. I presume other changes will be made outside the compiler to utilize it.

File: src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets [hunk 2]
```diff
@@ -50,8 +50,9 @@
     </PropertyGroup>
 
     <PropertyGroup>
+      <MaxSupportedLangVersion Condition="'$(MaxSupportedLangVersion)' == ''">7.3</MaxSupportedLangVersion>
```
Comment (333fred) [inline src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets]: This still seems wrong. This is explicitly only true when targeting older versions of netcore or framework.

Comment (agocke) [reply src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets]: Ah, yeah, you're right, I need to lift the condition out.

**Comment by 333fred** [comment]
@agocke Do we need to set MaxSupportedLangVersion for the non-7.3 case?

**Comment by agocke** [comment]
Nope, unset should just imply the max is whatevers in box

**Comment by agocke** [comment]
@davkean Did you ever find the right place to put this?

**Comment by davkean** [comment]
This should be outside of the target especially given https://github.com/dotnet/project-system/pull/5028 which only reads evaluation and will never see the above value.

**Pull request merged**